lib/ReactViews/Map/Panels/SettingPanel.tsx

```diff
@@ -36,384 +36,347 @@ const sides = {
-    event: MouseEvent<HTMLButtonElement>
-  ) {
-    const mainViewer = this.props.terria.mainViewer;
-    event.stopPropagation();
-    this.showTerrainOnSide(sides.both, undefined);
-    setViewerMode(viewer, mainViewer);
-    // We store the user's chosen viewer mode for future use.
-    this.props.terria.setLocalProperty("viewermode", viewer);
-    this.props.terria.currentViewer.notifyRepaintRequired();
-  }
+  const showTerrainOnSide = action(
+    (side: any, event?: MouseEvent<HTMLButtonElement>) => {
+      event?.stopPropagation();
 
-  @action
-  showTerrainOnSide(side: any, event?: MouseEvent<HTMLButtonElement>) {
-    event?.stopPropagation();
+      switch (side) {
+        case sides.left:
+          terria.terrainSplitDirection = SplitDirection.LEFT;
+          terria.showSplitter = true;
+          break;
+        case sides.right:
+          terria.terrainSplitDirection = SplitDirection.RIGHT;
+          terria.showSplitter = true;
+          break;
+        case sides.both:
+          terria.terrainSplitDirection = SplitDirection.NONE;
+          break;
+      }
 
-    switch (side) {
-      case sides.left:
-        this.props.terria.terrainSplitDirection = SplitDirection.LEFT;
-        this.props.terria.showSplitter = true;
-        break;
-      case sides.right:
-        this.props.terria.terrainSplitDirection = SplitDirection.RIGHT;
-        this.props.terria.showSplitter = true;
-        break;
-      case sides.both:
-        this.props.terria.terrainSplitDirection = SplitDirection.NONE;
-        break;
+      terria.currentViewer.notifyRepaintRequired();
+    }
+  );
+
+  const toggleDepthTestAgainstTerrainEnabled = action(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      event.stopPropagation();
+      terria.depthTestAgainstTerrainEnabled =
+        !terria.depthTestAgainstTerrainEnabled;
+      terria.currentViewer.notifyRepaintRequired();
     }
+  );
 
-    this.props.terria.currentViewer.notifyRepaintRequired();
-  }
+  const onBaseMaximumScreenSpaceErrorChange = (bmsse: number) => {
+    terria.setBaseMaximumScreenSpaceError(bmsse);
+    terria.setLocalProperty("baseMaximumScreenSpaceError", bmsse.toString());
+  };
 
-  @action
-  toggleDepthTestAgainstTerrainEnabled(event: ChangeEvent<HTMLInputElement>) {
-    event.stopPropagation();
-    this.props.terria.depthTestAgainstTerrainEnabled =
-      !this.props.terria.depthTestAgainstTerrainEnabled;
-    this.props.terria.currentViewer.notifyRepaintRequired();
-  }
+  const toggleUseNativeResolution = () => {
+    terria.setUseNativeResolution(!terria.useNativeResolution);
+    terria.setLocalProperty("useNativeResolution", terria.useNativeResolution);
+  };
 
-  onBaseMaximumScreenSpaceErrorChange(bmsse: number) {
-    this.props.terria.setBaseMaximumScreenSpaceError(bmsse);
-    this.props.terria.setLocalProperty(
-      "baseMaximumScreenSpaceError",
-      bmsse.toString()
-    );
+  if (!terria.mainViewer) {
+    return null;
```

Comment:
Looks like a leftover from past. `mainViewer` should always be defined. Can we remove it?